cumulusci/utils/__init__.py

```diff
@@ -677,11 +684,54 @@ def stderr_write(s):
     try:
         yield
     finally:
+        # reset write functions
         sys.stdout.write = real_stdout_write
         sys.stderr.write = real_stderr_write
 
+        # close temporary logfile
+        logger.handlers[0].close()
+
+        # log contents of tempfile to rotating log files
+        with open(tempfile, "r") as f:
+            contents = f.read()
+
+        logger = get_gist_logger()
+        logger.debug(contents)
+        # delete temporary log file
+        os.remove(tempfile)
+
+
+def get_gist_logger():
+    """Determines the appropriate filepath for logfile
+    and name for the logger. Returns a logger with
+    RotatingFileHandler attached."""
+    logfile_dir = Path.home() / ".cumulusci" / "logs"
+    logfile_dir.mkdir(parents=True, exist_ok=True)
+    logfile_path = logfile_dir / "cci.log"
+
+    return get_rot_file_logger("stdout/stderr", logfile_path)
+
+
+def get_rot_file_logger(name, path):
+    """Returns a logger with a rotating file handler"""
+    logger = logging.getLogger(name)
+
+    handler = RotatingFileHandler(path, backupCount=5, encoding="utf-8")
+    handler.doRollover()  # rollover existing log files
+    handler.terminator = ""  # click.echo already adds a newline
+    logger.addHandler(handler)
+    logger.setLevel(logging.DEBUG)
+    return logger
```

Comment:
Let's make a new module called logging.py in cumulusci/utils and move the logging-related utils there. Over time I want to split all of utils/__init__.py into separate functional groups like this.